src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -1424,14 +1489,16 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
                     let local_version = local_source_map.new_imported_source_file(
                         name,
                         name_was_remapped,
-                        self.cnum.as_u32(),
                         src_hash,
                         name_hash,
                         source_length,
+                        self.cnum,
```

Comment:
`cnum` is now encoded/decoded and can be taken from `source_file_to_import` above but here it's still taken from `CrateMetadata`.
So it seems like we are encoding/decoding it in vain.

Reply:
We need to decode `CrateNum` so that we know which `CrateMetadata` to call `imported_source_files` on. We could pass in the `CrateNum` to `imported_source_files`, but I think it's simpler just to use `self.cnum`.